tests/integration/server/faculty/facultyapi.test.ts

```diff
@@ -0,0 +1,1062 @@
+              .createQueryBuilder('a')
+              .leftJoinAndMapOne(
+                'a.semester',
+                Semester, 's',
+                'a."semesterId" = s."id"'
+              )
+              .where(
+                's."academicYear"=:acyr',
+                {
+                  acyr: thisAcademicYear,
+                  term: TERM.SPRING,
+                }
+              )
+              .getOne();
+
+            const { status } = await request(api)
+              .put(`/api/faculty/absence/${springAbsenceThisYear.id}`)
+              .send({
+                id: springAbsenceThisYear.id,
+                type: ABSENCE_TYPE.TEACHING_RELIEF,
+              });
+
+            const { updatedAt } = await absenceRepository.findOne(
+              springAbsenceThisYear.id,
+              { select: ['updatedAt'] }
+            );
+            strictEqual(status, HttpStatus.OK);
+            notStrictEqual(updatedAt, springAbsenceThisYear.updatedAt);
+          });
+        });
+        describe('absences in future academic years', function () {
+          it('allows modification of absences after the current academic year', async function () {
+            const { status } = await request(api)
+              .put(`/api/faculty/absence/${absenceNextYear.id}`)
+              .send({
+                id: absenceNextYear.id,
+                type: ABSENCE_TYPE.TEACHING_RELIEF,
+              });
+
+            const {
+              updatedAt,
+            } = await absenceRepository.findOne(
+              absenceNextYear.id,
+              { select: ['updatedAt'] }
+            );
+            strictEqual(status, HttpStatus.OK);
+            notDeepStrictEqual(updatedAt, absenceNextYear.updatedAt);
+          });
+        });
+        describe(`changing FROM ${absenceEnumToTitleCase(ABSENCE_TYPE.NO_LONGER_ACTIVE)}`, function () {
+          beforeEach(async function () {
+            // Update the absence to be longer active
+            await absenceRepository.createQueryBuilder('a')
+              .update(Absence)
+              .set({ type: ABSENCE_TYPE.NO_LONGER_ACTIVE })
+              .where({ id: In([springAbsence.id, fallAbsence.id]) })
+              .execute();
+          });
+          it('does not modify past absences', async function () {
+            const absencesBeforeUpdate = (await absenceRepository.find({
+              select: ['type', 'id'],
+              relations: ['semester'],
+              where: { faculty: springAbsence.faculty.id },
+              order: {
+                id: 'ASC',
+              },
+            }))
+              .filter(({ semester }) => parseInt(semester.academicYear, 10)
+                < thisAcademicYear);
+
+            await request(api)
+              .put(`/api/faculty/absence/${springAbsence.id}`)
+              .send({
+                id: springAbsence.id,
+                type: ABSENCE_TYPE.TEACHING_RELIEF,
+              });
```

Comment:
Missing a test to ensure that the requested semester actually gets updated (see note on `FacultyService`)

Reply:
Added a test for this in 4a2ddc0b34a1c94f1caf9fc2f57c03801b7a7281